packages/vite/src/node/server/middlewares/transform.ts

```diff
@@ -205,12 +205,19 @@ export function transformMiddleware(
           const type = isDirectCSSRequest(url) ? 'css' : 'js'
           const isDep =
             DEP_VERSION_RE.test(url) || depsOptimizer?.isOptimizedDepUrl(url)
+          const filepath =
+            result.map == null
+              ? (await server.moduleGraph.getModuleByUrl(url, false))?.file
+              : undefined
+          const originalContent =
+            filepath != null ? await getOriginalContent(filepath) : undefined
```

Comment:
Maybe something like this?
```suggestion
          let originalContent: string | undefined
          if (type === 'js' && result.map == null) {
            const filepath =
              result.map == null
                ? (await server.moduleGraph.getModuleByUrl(url, false))?.file
                : undefined
            originalContent =
              filepath != null ? await getOriginalContent(filepath) : undefined
          }
```

Reply:
Ah, I see now that you had the condition directly in `filepath` so it won't be actually triggered as much as I thought. So yes, only adding `type === 'js'`. Maybe a direct condition would have been easier to read.